runtime/src/iree/hal/utils/mpi_channel_provider.c

```diff
@@ -68,10 +68,15 @@ IREE_API_EXPORT iree_status_t iree_hal_mpi_channel_provider_create(
 
   // If the library successfully loaded then try to initialize MPI.
   if (iree_status_is_ok(status)) {
-    IREE_TRACE_ZONE_BEGIN_NAMED(z1, "MPI_Init");
-    status = MPI_RESULT_TO_STATUS(&channel_provider->symbols,
-                                  MPI_Init(NULL, NULL), "MPI_Init");
-    IREE_TRACE_ZONE_END(z1);
+    int flag = 0;
+    MPI_IGNORE_ERROR(&channel_provider->symbols, MPI_Initialized(&flag));
+
+    if (!flag) {
+      IREE_TRACE_ZONE_BEGIN_NAMED(z1, "MPI_Init");
```

Comment:
this will need to track whether it did the init or not (`bool did_initialize;` or something) so that the MPI_Finalize can be run only if it was actually initialized - otherwise destroying the provider will finalize even if the hosting application didn't expect it to

Reply:
I think I can retain the channel provider and release only when the hosting layer is destructed. I will try this first.